libafl/src/executors/forkserver.rs

```diff
@@ -488,27 +488,25 @@ impl Forkserver {
     }
 
     /// Read from the st pipe
-    pub fn read_st(&mut self) -> Result<(usize, i32), Error> {
+    pub fn read_st(&mut self) -> Result<i32, Error> {
         let mut buf: [u8; 4] = [0_u8; 4];
-
-        let rlen = self.st_pipe.read(&mut buf)?;
+        self.st_pipe.read_exact(&mut buf)?;
```

Comment:
Same here: We should `.read` but check for `length != 4` and return a result if it failed.
It's a minimal optimization, but we're in a hot path here.